src/net/gosha/atproto/client.clj

```diff
@@ -1,39 +1,101 @@
 (ns net.gosha.atproto.client
   (:require
+   [clojure.spec.alpha :as s]
    [martian.core :as martian]
-   [martian.httpkit :as martian-http]
-   [net.gosha.atproto.core :as core]))
+   [martian.httpkit :as martian-http]))
 
-(def openapi-url "https://raw.githubusercontent.com/bluesky-social/bsky-docs/refs/heads/main/atproto-openapi-types/spec/api.json")
+(s/def ::auth-token string?)
+(s/def ::username string?)
+(s/def ::base-url string?)
+(s/def ::app-password string?)
+(s/def ::openapi-spec string?)
+(s/def ::config (s/keys :req-un [::base-url ::openapi-spec]
+                  :opt-un [::auth-token ::app-password ::username]))
 
-(defn add-authentication-header [token]
+(def ^{:doc "map of config keys to env vars that can set them"}
+  env-keys
+  {:openapi-spec "ATPROTO_OPENAPI_SPEC"
+   :base-url "ATPROTO_BASE_URL"
+   :auth-token "ATPROTO_AUTH_TOKEN"
+   :username "ATPROTO_USERNAME"
+   :app-password "ATPROTO_APP_PASSWORD"})
+
+(defn- build-config
+  "Build a configuration map based on defaults, env vars and provided values"
+  [& {:as user-config}]
+  (-> {:openapi-spec "./resources/atproto-xrpc-openapi.2024-12-18.json"}
```

Comment:
This might be something to worry about later, but if we're going to store the json locally, we should have some mechanism for reliably updating it, possibly through a `make` task that we could run via github actions or some other automation.